ext/standard/basic_functions.c

```diff
@@ -896,38 +896,33 @@ PHP_FUNCTION(putenv)
 		unsetenv(pe.putenv_string);
 	}
 	if (!p || putenv(pe.putenv_string) == 0) { /* success */
-#else
-# ifndef PHP_WIN32
+#elif !defined(PHP_WIN32)
 	if (putenv(pe.putenv_string) == 0) { /* success */
-# else
-		wchar_t *keyw, *valw = NULL;
-
-		keyw = php_win32_cp_any_to_w(pe.key);
-		if (value) {
-			valw = php_win32_cp_any_to_w(value);
-		}
-		/* valw may be NULL, but the failed conversion still needs to be checked. */
-		if (!keyw || !valw && value) {
-			efree(pe.putenv_string);
-			efree(pe.key);
-			free(keyw);
-			free(valw);
-			RETURN_FALSE;
-		}
+#else
+	wchar_t *keyw, *valw = NULL;
 
-	error_code = SetEnvironmentVariableW(keyw, valw);
+	keyw = php_win32_cp_any_to_w(pe.key);
+	if (value) {
+		valw = php_win32_cp_any_to_w(value);
+	}
+	/* valw may be NULL, but the failed conversion still needs to be checked. */
+	if (!keyw || !valw && value) {
+		efree(pe.putenv_string);
+		efree(pe.key);
+		free(keyw);
+		free(valw);
```

Comment:
This whole code is a big mess and we fail to unlock in this branch.

Reply:
Thanks! I fixed the latter with 32257ac17f68b4f8a5c276b92521f338f6c7cf86 (PHP-7.4), and will have a look at cleaning up the code as soon as possible.